scalahost-sbt/src/main/scala/scala/meta/scalahost/sbt/ScalahostSbtPlugin.scala

```diff
@@ -46,21 +59,75 @@ object ScalahostSbtPlugin extends AutoPlugin {
               )
             }
           List(
-            s"-Xplugin:${scalahostJar}",
+            s"-Xplugin:$scalahostJar",
             "-Yrangepos",
             "-Xplugin-require:scalahost"
           )
         case _ => Nil
       }
+    }
+  )
+
+  lazy val scalahostHasMirrorSettings: Seq[Def.Setting[_]] = Def.settings(
+    // Automatically set scalahostAggregate to projects listed in dependsOn(x % Scalahost)
+    scalahostAggregate := thisProject.value.dependencies.collect {
+      case x if x.configuration.exists(_ == Scalahost.name) => x.project
+    },
+    javaOptions ++= {
+      if (scalahostAggregate.value.isEmpty) Nil
+      else {
+        val sourcepath =
+          scalahostSourcepath.value
+            .map(_.getAbsolutePath)
+            .mkString(",")
+        val classpath =
+          scalahostClasspath.value
+            .flatMap(_.files.map(_.getAbsolutePath))
+            .mkString(java.io.File.pathSeparator)
+        List(
+          s"-Dscalameta.mirror.sourcepath=$sourcepath",
+          s"-Dscalameta.mirror.classpath=$classpath"
+        )
+      }
+    },
+    // fork := true is required to pass along -Dscalameta.mirror.{classpath,sourcepath}
+    fork in run := {
+      if (scalahostAggregate.value.isEmpty) fork.in(run).value
+      else true
     },
+    // automatically depend on scalahost if this project dependsOn(otherProject % Scalahost)
     libraryDependencies ++= {
-      scalaVersion.value match {
-        case SupportedScalaVersion(version) =>
-          List(
-            "org.scalameta" % s"scalahost_$version" % scalahostVersion % MetaConfig
-          )
-        case _ => Nil
-      }
+      if (scalahostAggregate.value.isEmpty) Nil
+      else
+        List(
+          ("org.scalameta" % "scalahost" % scalahostVersion).cross(CrossVersion.full)
+        )
     }
   )
+
+  // Scalahost only supports the latest patch version of scala minor version, for example
+  // 2.11.8 and not 2.11.7. If a build is using 2.11.7, we upgrade the dependency to
+  // scalahost 2.11.8. See https://github.com/scalameta/scalameta/issues/681
+  private object SupportedScalaVersion {
+    private val MajorMinor = "(\\d+\\.\\d+)..*".r
+    def unapply(arg: String): Option[String] = arg match {
+      case MajorMinor(version) =>
+        BuildInfo.supportedScalaVersions.find(_.startsWith(version))
+      case _ => None
+    }
+  }
+  // Defaults to version.value of in scala.meta's build.sbt
+  private val scalahostVersion: String =
+    sys.props.getOrElse("scalahost.version", BuildInfo.version)
```

Comment:
I use this all the time for local development on sbt-scalahost. Without it, I would need to republish all packages on every build reload.